packages/curve-ui-kit/src/shared/ui/Metric.tsx

```diff
@@ -0,0 +1,232 @@
+type Unit = (typeof UNITS)[number] | UnitOptions
+
+const UNIT_MAP: Record<(typeof UNITS)[number], UnitOptions> = {
+  dollar,
+  percentage,
+} as const
+
+// Default value formatter.
+const formatValue = (value: number, decimals?: number): string =>
+  value.toLocaleString(undefined, {
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
+  })
+
+// Default notional value formatter.
+const formatNotionalValue = (value: number, decimals?: number): string =>
+  value.toLocaleString(undefined, {
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
+  })
+
+const formatChange = (value: number) => {
+  // Looks aesthetically more pleasing without decimals.
+  if (value === 0) return '0'
+
+  return value.toLocaleString(undefined, {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  })
+}
+
+type Props = {
+  /** Label that goes above the value */
+  label: string
+  /** Optional tooltip content shown next to the label */
+  tooltip?: string
+
+  /** The actual metric value to display */
+  value: number
+  /** Optional formatter for metric value */
+  formatter?: (value: number) => string
+  /** If the value should be abbreviated to 1.23k or 3.45m */
+  abbreviate?: boolean
+  /** The number of decimals the value should contain */
+  decimals?: number
+  /** A unit can be a currency symbol or percentage, prefix or suffix */
+  unit?: Unit
+
+  /** Optional value that denotes a change in metric value since 'last' time */
+  change?: number
+
+  /** Notional values give extra context to the metric, like underlying value */
+  notional?: number
+  notionalFormatter?: (value: number) => string
+  notionalAbbreviate?: boolean
+  notionalDecimals?: number
+  notionalUnit?: Unit
+
+  size?: keyof typeof MetricSize
+  alignment?: Alignment
+  loading?: boolean
+}
+
+export const Metric = ({
+  label,
+  tooltip,
+
+  value,
+  formatter = (value: number) => formatValue(value, decimals),
+  abbreviate,
+  unit,
+  decimals = 1,
+
+  change,
+
+  notional,
+  notionalFormatter = (value: number) => formatNotionalValue(value, notionalDecimals),
+  notionalAbbreviate,
+  notionalUnit,
+  notionalDecimals,
+
+  size = 'medium',
+  alignment = 'start',
+  loading = false,
+}: Props) => {
+  unit = typeof unit === 'string' ? UNIT_MAP[unit] : unit
+  abbreviate ??= unit?.abbreviate ?? false
+
+  notionalUnit = typeof notionalUnit === 'string' ? UNIT_MAP[notionalUnit] : notionalUnit
+  notionalAbbreviate ??= notionalUnit?.abbreviate ?? false
+
+  const [openCopyAlert, setOpenCopyAlert] = useState(false)
+
+  const copyValue = () => {
+    navigator.clipboard.writeText(value.toString())
+    setOpenCopyAlert(true)
+  }
+
+  return (
+    <Box display="flex" flexDirection="column" alignItems={alignment} gap={Spacing.xs}>
+      <Typography variant="bodyXsRegular" color="textTertiary">
+        {label}
+        {tooltip && (
+          <Tooltip title={tooltip}>
+            <span>
+              &nbsp;
```

Comment:
I only wanted to add a space when the tooltip was shown, and wanted exactly the size of "one spacebar" which I couldn't replace with margin (1ch is wider than a space). Had to wrap it in a span because Tooltip doesn't accept text as children, and `<></>` would break stuff.

Reply:
This might be a bit clearer
```suggestion
              {' '}
```